netweaver/monitoring.sls

```diff
@@ -20,17 +25,22 @@ sap_host_exporter_configuration_{{ exporter_instance }}:
          sap-control-uds: /tmp/.sapstream5{{ sap_instance_nr }}13
     - require:
       - pkg: prometheus_sap_host_exporter_pkg
-      - netweaver_install_{{ instance_name }}
 
+# do not run ASCS and ERS exporter service in HA use case (handled by pacemaker)
+# only run PAS exporter in HA use case (if no dedicated PAS host is used)
+{% if not netweaver.ha_enabled or node.sap_instance == 'pas' %}
```

Comment:
This looks pretty risky. We are assuming that the HA cluster formula will be executed later, and with this the exporter resource agent. I wouldn't take this as granted.
I would just prefer to start the exporter with the `enable` as before.
What was the risk of doing that?

Finally, why is the AAS excluded here?

Reply:
I wanted to build it the safe way and not interfere with the HA cluster formula (and pacemaker running the systemd resource) at all. In case of a systemd resource this is ok...as pacemaker will just run a probe/monitor and see that the systemd service is already up even if it fails, it will not stop the SAP instance, as it is started later in the group.
How is this "do not touch" approach more risky than running a service that is handled by the cluster?

AAS was excluded as it is supposed to never run on an HA node. It is included in the `if not netweaver.ha_enabled` statement.
But I think now after reviewing it that I got `netweaver.ha_enabled` wrong... This is global and also set on PAS/AAS.

A simple `enable: {{ not netweaver.ha_enabled }}` does disable the service also on PAS and AAS, doesn't it (if ASCS/ERS are HA)? So it will not come up after a reboot.

Another problem is, that `- require: - netweaver_install_{{ instance_name }}` cannot be used this way as ASCS/ERS will not be installed on ERS/ASCS nodes.
Do you think it is okay to remove the requirement at all? Initially I already removed it from the config file section as the config files are needed on both HA nodes.
But than again... one has to consider starting the ASCS/ERS exporter services only on one side and not on both. Also starting exporters on not running instances is bad.
The way to go is most likely just running `netweaver.monitoring` after `netweaver.install_*` which should already be the case.

Reply:
This could be a hybrid approach.

```
# do not enable ASCS and ERS exporter service on in HA use case (handled by pacemaker)
{% if netweaver.ha_enabled and node.sap_instance in ['ascs', 'ers'] %}
{% set exporter_running = False %}
{% else %}
{% set exporter_running = True %}
{% endif %}

sap_host_exporter_service_{{ exporter_instance }}:
{% if exporter_running %}
  service.running:
{% else %}
  service.dead:
{% endif %}
    - name: prometheus-sap_host_exporter@{{ exporter_instance }}
    - enable: {{ exporter_running }}
[...]
```

Reply:
@arbulu89 fixed in 4ffafae